lib/backend/firestore/firestorebk.go

```diff
@@ -567,6 +579,72 @@ func (b *Backend) Delete(ctx context.Context, key []byte) error {
 	return nil
 }
 
+// ConditionalDelete deletes item by key if the revision matches
+func (b *Backend) ConditionalDelete(ctx context.Context, key []byte, rev string) error {
+	if len(key) == 0 {
```

Comment:
Should we enforce non-empty revision or is it here for compatibility reasons for versions without a revision field?
if it's the last, can we add a comment to mention it?